.circleci/config.yml

```diff
@@ -0,0 +1,184 @@
+          name: Git save cache
+          key: source-v1-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - ".git"
+  npm_install:
+    description: "Install npm modules"
+    steps:
+      - restore_cache:
+          name: Restore npm cache
+          keys:
+            - npm-v1-{{ checksum "package.json" }}
+            - npm-v1-
+      - run:
+          name: Install npm modules
+          command: npm install && ./node_modules/.bin/bower install
+      - save_cache:
+          name: Save NPM cache
+          key: npm-v1-{{ checksum "package.json" }}
+          paths:
+            - "node_modules"
+  build:
+    description: "Build"
+    steps:
+      - run:
+          name: "gulp build"
+          command: ./node_modules/.bin/gulp build
+  compress:
+    description: "Compress"
+    steps:
+      - run:
+          name: "Compress"
+          command: |
+            pushd www/
+            tar -cvf artifact.tar *
+            mv artifact.tar ${OLDPWD}/
+      - run:
+          name: "Tag commit id as artifact identifer"
+          command: echo "${CIRCLE_SHA1}" > artifact-info.txt
+  upload_artifact:
+    description: "upload artifact to s3"
+    steps:
+      - s3/copy:
+          from: artifact.tar
+          to: 's3://${CONTEXT_ARTIFACT_S3_BUCKET}/${CIRCLE_PROJECT_REPONAME}/'
+          aws-access-key-id: env_CONTEXT_ARTIFACT_S3_AWS_ACCESS_KEY_ID
+          aws-secret-access-key: env_CONTEXT_ARTIFACT_S3_AWS_SECRET_ACCESS_KEY
+          aws-region: env_CONTEXT_ARTIFACT_S3_AWS_REGION
+          arguments: '--metadata "{\"x-amz-artifact-id\": \"${CIRCLE_SHA1}\" }"'
+  upload_checksum:
+    description: "upload artifact commit id to s3"
+    steps:
+      - s3/copy:
+          from: artifact-info.txt
+          to: 's3://${CONTEXT_ARTIFACT_S3_BUCKET}/${CIRCLE_PROJECT_REPONAME}/'
+          aws-access-key-id: env_CONTEXT_ARTIFACT_S3_AWS_ACCESS_KEY_ID
+          aws-secret-access-key: env_CONTEXT_ARTIFACT_S3_AWS_SECRET_ACCESS_KEY
+          aws-region: env_CONTEXT_ARTIFACT_S3_AWS_REGION
+  docker:
+    description: "Build and Push image to docker hub"
+    parameters:
+      target:
+        type: string
+    steps:
+      - setup_remote_docker
+      - run:
+          name: Building docker image for << parameters.target >>
+          command: |
+            build_tag="${CIRCLE_SHA1}"
+            [ "<< parameters.target >>" == "beta" ] && build_tag="beta-${CIRCLE_SHA1}"
+            docker build -t ${DOCKHUB_ORGANISATION}/ticktrade:${build_tag} -t ${DOCKHUB_ORGANISATION}/ticktrade:latest .
+      - run:
+          name: Pushing Image to docker hub
+          command: |
+            build_tag="${CIRCLE_SHA1}"
+            [ "<< parameters.target >>" == "beta" ] && build_tag="beta-${CIRCLE_SHA1}"
+            echo $DOCKERHUB_PASSWORD | docker login -u $DOCKERHUB_USERNAME --password-stdin
+            docker push ${DOCKHUB_ORGANISATION}/ticktrade:${build_tag}
+            docker push ${DOCKHUB_ORGANISATION}/ticktrade:latest
```

Comment:
with one docker push command you can upload all the separate tagged image in a single docker hub repository. you can get some idea from here, https://github.com/shudarshon-binary/deriv-app/blob/chaks/circle-artifact-upload/.circleci/config.yml

Reply:
the great command to know, It's fixed.